### PR TITLE
Sane numpy

### PR DESCRIPTION
**Pull request by kain88-de**
switch to a ABI compatible numpy configuration and add python 3.4 to build matrix.

### CODE REVIEW

**Comment by peterjc** [comment]
I assume this is part of a general conda-forge numpy update effort (the details of I do not understand and am not commenting on).

Good catch on checking importing of ``Bio.Cluster`` and ``Bio.KDTree`` in the tests which are both dependent on compiled C code.

Since you are working on this, can you also add ``Bio.Affy``, ``Bio.phenotype``, ``Bio.cpairwise2``, ``Bio.trie`` to the import tests as well (again C code, but not all using NumPy itself).

Cross reference https://github.com/bioconda/bioconda-recipes/pull/5465 for fixing the same oversight in the older BioConda recipe which this conda-forge recipe was based on.

**Comment by kain88-de** [comment]
@ocefpaf there seems to be a numpy/python issue on windows with python 3.5. 



> I assume this is part of a general conda-forge numpy update effort (the details of I do not understand and am not commenting on).

Yes the idea is that we can update numpy without having to rebuild all packages depending on numpy when a new numpy version is released.
see: https://github.com/conda-forge/conda-forge.github.io/pull/415

**Comment by peterjc** [comment]
@kain88-de Understood, thank you. We're trying to do something similar with the Biopython wheels for PyPI where we build against the oldest numpy as possible for each Python, so our wheels should work with any more recent numpy: https://github.com/biopython/biopython-wheels

**Comment by ocefpaf** [comment]
> @ocefpaf there seems to be a numpy/python issue on windows with python 3.5.

That is my bad. When I wrote the docs I was sure I saw `numpy 1.8` for `Python 3.5` in `defaults`. We need to use `1.9` there :unamused: 

I am working to get `numpy` for Windows on conda-forge so we can have control over this.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -12,30 +12,36 @@ source:
...
   script: python setup.py install --single-version-externally-managed --record record.txt
 
 requirements:
   build:
     - python
     - setuptools
-    - numpy x.x
+    - numpy 1.8.*  # [not (win and (py35 or py36))]
+    - numpy 1.9.*  # [win and py35]
+    - numpy 1.11.*  # [win and py36]
```
Comment (ocefpaf) [inline recipe/meta.yaml:24]: @kain88-de I guess that this is the current combination. I don't really want to spread this at the moment. I'd rather work fast to get older versions of `numpy` in the channel so we can simplify this ASAP.

Comment (kain88-de) [reply recipe/meta.yaml:24]: I can wait for numpy 1.8 on windows. I rather have a correct long-term solution for the maintainers. I also changed these to see if I can break things in the build ;-)

Comment (ocefpaf) [reply recipe/meta.yaml:24]: Cool. We'll have a meeting next Friday and I'll try to ask for a numpy 1.8 and 1.9 on `defaults` for both py35 and py36.

**Comment by souravsingh** [comment]
@kain88-de Is the patch ready for merging or should we wait until NumPy 1.8 and 1.9 is available in the default channel?

**Comment by kain88-de** [comment]
@ocefpaf how are the default channels looking?

@souravsingh depends on you. If we wait you don't need to change this in the future.

**Comment by ocefpaf** [comment]
> @ocefpaf how are the default channels looking?

Promising. We discussed that last meeting and people are looking into it. I'll know more next week.

**Comment by kain88-de** [comment]
@ocefpaf any change on this?

**Comment by ocefpaf** [comment]
> @ocefpaf any change on this?

Nope. I'll ask again today but maybe we should not wait for that...

**Comment by kain88-de** [comment]
@souravsingh You can merge this if you like.

**Comment by kain88-de** [comment]
The merge is currently failing because there is an issue with the anaconda token.

**Comment by kain88-de** [comment]
https://github.com/conda-forge/conda-forge.github.io/issues/426#issuecomment-321897876

looks like we can wait until there pops up a PR to fix the issue.

**Comment by ocefpaf** [comment]
The short-term solution is to re-submit the recipe to staged-recipes. It won't change the recipe here, only the tokens.

**Comment by souravsingh** [comment]
@ocefpaf Do I now have to submit a PR to staged-recipes with the new recipe? If so, What would happen to this feedstock repository?

**Pull request merged**